src/main/java/net/dv8tion/jda/internal/requests/WebSocketClient.java

```diff
@@ -837,6 +840,19 @@ protected void onDispatch(DataObject raw)
                         jda.setStatus(JDA.Status.LOADING_SUBSYSTEMS);
                     }
                     break;
+                case "GUILD_MEMBERS_CHUNK":
+                    synchronized (memberRequests)
```

Comment:
While this will prevent the modification of `memberRequests` while the payload is being handled, it doesn't handle the fact that the following situation could occur:

1) request presence for Id 123
2) request presence for Id 234
3) payload response received for 123
 3.a) callbacks run.
 3.b) callback for 123 succeeds
 3.c) callback for 234 gets empty presence?
 3.d) all callbacks removed
4) payload response received for 234
 4.a) no callback to handle 234

Am I missing something?

Reply:
I'm still working on this and I know its incomplete. I'm trying to think of a good way to implement this.

Reply:
What of we used a range from 0 - 100 or so to represent unique identifiable member chunk requests instead of just 0?

We could use a different number for each one so that we only run and clear a callback when we get a payload response for the one we requested.